### PR TITLE
Add alert words API

### PR DESCRIPTION
**Pull request by youknowwjk**
Supplement the API for the alert words section, which is missing from the project

### CODE REVIEW

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java [hunk 1]
```diff
@@ -1,32 +1,8 @@
...
-import com.github.jamesnetherton.zulip.client.api.user.request.RemoveUsersFromGroupApiRequest;
-import com.github.jamesnetherton.zulip.client.api.user.request.SetTypingStatusApiRequest;
-import com.github.jamesnetherton.zulip.client.api.user.request.UnmuteUserApiRequest;
-import com.github.jamesnetherton.zulip.client.api.user.request.UpdateNotificationSettingsApiRequest;
-import com.github.jamesnetherton.zulip.client.api.user.request.UpdateOwnUserSettingsApiRequest;
-import com.github.jamesnetherton.zulip.client.api.user.request.UpdateOwnUserStatusApiRequest;
-import com.github.jamesnetherton.zulip.client.api.user.request.UpdateUserApiRequest;
-import com.github.jamesnetherton.zulip.client.api.user.request.UpdateUserGroupApiRequest;
-import com.github.jamesnetherton.zulip.client.api.user.request.UpdateUserGroupSubGroupsApiRequest;
+import com.github.jamesnetherton.zulip.client.api.user.request.*;
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java:4]: Please can you avoid import `*` and reinstate the individual imports.

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java [hunk 2]
```diff
@@ -404,4 +380,42 @@ public GetSubGroupsOfUserGroupApiRequest getSubGroupsOfUserGroup(long userGroupI
     public UpdateUserGroupSubGroupsApiRequest updateUserGroupSubGroups(long userGroupId) {
         return new UpdateUserGroupSubGroupsApiRequest(this.client, userGroupId);
     }
+
+    /**
+     * add-alert-words
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java:385]:
```suggestion
     * Adds alert words.
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java [hunk 3]
```diff
@@ -404,4 +380,42 @@ public GetSubGroupsOfUserGroupApiRequest getSubGroupsOfUserGroup(long userGroupI
...
+     *
+     * @param  alertWords An array of strings, where each string is an alert word (or phrase)
+     * @return             The {@link AddAlertWordsApiRequest} builder object
+     */
+    public AddAlertWordsApiRequest addAlertWords(String[] alertWords) {
+        return new AddAlertWordsApiRequest(this.client, alertWords);
+    }
+
+    /**
+     * get all alert words
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java:398]:
```suggestion
     * Gets all alert words.
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java [hunk 4]
```diff
@@ -404,4 +380,42 @@ public GetSubGroupsOfUserGroupApiRequest getSubGroupsOfUserGroup(long userGroupI
...
+     *                     "https://biocypher.zulipchat.com/api/v1/users/me/alert_words">https://biocypher.zulipchat.com/api/v1/users/me/alert_words</a>
+     *
+     * @return             The {@link AddAlertWordsApiRequest} builder object
+     */
+    public GetAllAlertWordsApiRequest getAllAlertWords() {
+        return new GetAllAlertWordsApiRequest(this.client);
+    }
+
+    /**
+     * remove-alert-words
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java:410]:
```suggestion
     * Removes alert words.
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java [hunk 5]
```diff
@@ -404,4 +380,42 @@ public GetSubGroupsOfUserGroupApiRequest getSubGroupsOfUserGroup(long userGroupI
     public UpdateUserGroupSubGroupsApiRequest updateUserGroupSubGroups(long userGroupId) {
         return new UpdateUserGroupSubGroupsApiRequest(this.client, userGroupId);
     }
+
+    /**
+     * add-alert-words
+     *
+     * @see                <a href=
+     *                     "https://biocypher.zulipchat.com/api/v1/users/me/alert_words">https://biocypher.zulipchat.com/api/v1/users/me/alert_words</a>
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java:388]:
```suggestion
     *                     "https://zulip.com/api/add-alert-words">https://zulip.com/api/add-alert-words</a>
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java [hunk 6]
```diff
@@ -404,4 +380,42 @@ public GetSubGroupsOfUserGroupApiRequest getSubGroupsOfUserGroup(long userGroupI
...
+     */
+    public AddAlertWordsApiRequest addAlertWords(String[] alertWords) {
+        return new AddAlertWordsApiRequest(this.client, alertWords);
+    }
+
+    /**
+     * get all alert words
+     *
+     * @see                <a href=
+     *                     "https://biocypher.zulipchat.com/api/v1/users/me/alert_words">https://biocypher.zulipchat.com/api/v1/users/me/alert_words</a>
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java:401]:
```suggestion
     *                     "https://zulip.com/api/get-alert-words">https://zulip.com/api/get-alert-words</a>
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java [hunk 7]
```diff
@@ -404,4 +380,42 @@ public GetSubGroupsOfUserGroupApiRequest getSubGroupsOfUserGroup(long userGroupI
...
+     *
+     * @return             The {@link AddAlertWordsApiRequest} builder object
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java:403]:
```suggestion
     * @return             The {@link GetAllAlertWordsApiRequest} builder object
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java [hunk 8]
```diff
@@ -404,4 +380,42 @@ public GetSubGroupsOfUserGroupApiRequest getSubGroupsOfUserGroup(long userGroupI
...
+     */
+    public GetAllAlertWordsApiRequest getAllAlertWords() {
+        return new GetAllAlertWordsApiRequest(this.client);
+    }
+
+    /**
+     * remove-alert-words
+     *
+     * @see                <a href=
+     *                     "https://biocypher.zulipchat.com/api/v1/users/me/alert_words">https://biocypher.zulipchat.com/api/v1/users/me/alert_words</a>
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java:413]:
```suggestion
     *                     "https://zulip.com/api/remove-alert-words">https://zulip.com/api/remove-alert-words</a>
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/AddAlertWordsApiRequest.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+import com.github.jamesnetherton.zulip.client.api.core.ExecutableApiRequest;
+import com.github.jamesnetherton.zulip.client.api.core.ZulipApiRequest;
+import com.github.jamesnetherton.zulip.client.api.user.response.GetAllAlertWordsApiResponse;
+import com.github.jamesnetherton.zulip.client.exception.ZulipClientException;
+import com.github.jamesnetherton.zulip.client.http.ZulipHttpClient;
+
+/**
+ * Zulip API request builder for set alert words.
+ *
+ * @see <a href="https://zulip.com/api/v1/users/me/alert_words">https://zulip.com/api/v1/users/me/alert_wordsr</a>
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/AddAlertWordsApiRequest.java:14]:
```suggestion
 * @see <a href="https://zulip.com/api/add-alert-words">https://zulip.com/api/add-alert-words</a>
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/GetAllAlertWordsApiRequest.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+import com.github.jamesnetherton.zulip.client.exception.ZulipClientException;
+import com.github.jamesnetherton.zulip.client.http.ZulipHttpClient;
+
+import static com.github.jamesnetherton.zulip.client.api.user.request.UserRequestConstants.USERS_WITH_ME;
+
+
+/**
+ * Zulip API request builder for Get all the user's configured alert words.
+ *
+ * @see <a href="https://zulip.com/api/v1/users/me/alert_words">https://zulip.com/api/v1/users/me/alert_wordsr</a>
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/GetAllAlertWordsApiRequest.java:15]:
```suggestion
 * @see <a href="https://zulip.com/api/get-alert-words">https://zulip.com/api/get-alert-words</a>
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/RemoveAlertWordsApiRequest.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+import com.github.jamesnetherton.zulip.client.api.user.response.GetAllAlertWordsApiResponse;
+import com.github.jamesnetherton.zulip.client.exception.ZulipClientException;
+import com.github.jamesnetherton.zulip.client.http.ZulipHttpClient;
+
+import static com.github.jamesnetherton.zulip.client.api.user.request.UserRequestConstants.USERS_WITH_ME;
+
+/**
+ * Zulip API request builder for remove alert words.
+ *
+ * @see <a href="https://zulip.com/api/v1/users/me/alert_words">https://zulip.com/api/v1/users/me/alert_wordsr</a>
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/RemoveAlertWordsApiRequest.java:14]:
```suggestion
 * @see <a href="https://zulip.com/api/remove-alert-words">https://zulip.com/api/remove-alert-words</a>
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java [hunk 12]
```diff
@@ -404,4 +380,42 @@ public GetSubGroupsOfUserGroupApiRequest getSubGroupsOfUserGroup(long userGroupI
...
+    /**
+     * add-alert-words
+     *
+     * @see                <a href=
+     *                     "https://biocypher.zulipchat.com/api/v1/users/me/alert_words">https://biocypher.zulipchat.com/api/v1/users/me/alert_words</a>
+     *
+     * @param  alertWords An array of strings, where each string is an alert word (or phrase)
+     * @return             The {@link AddAlertWordsApiRequest} builder object
+     */
+    public AddAlertWordsApiRequest addAlertWords(String[] alertWords) {
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java:393]:
```suggestion
    public AddAlertWordsApiRequest addAlertWords(String... alertWords) {
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java [hunk 13]
```diff
@@ -404,4 +380,42 @@ public GetSubGroupsOfUserGroupApiRequest getSubGroupsOfUserGroup(long userGroupI
...
+    /**
+     * remove-alert-words
+     *
+     * @see                <a href=
+     *                     "https://biocypher.zulipchat.com/api/v1/users/me/alert_words">https://biocypher.zulipchat.com/api/v1/users/me/alert_words</a>
+     *
+     * @param  alertWords An array of strings, where each string is an alert word (or phrase)
+     * @return             The {@link RemoveAlertWordsApiRequest} builder object
+     */
+    public RemoveAlertWordsApiRequest removeAlertWords(String[] alertWords) {
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java:418]:
```suggestion
    public RemoveAlertWordsApiRequest removeAlertWords(String... alertWords) {
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/AddAlertWordsApiRequest.java [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+    public static final String ALERT_WORDS = "alert_words";
+
+    /**
+     * Constructs a {@link AddAlertWordsApiRequest}.
+     *
+     * @param client     The Zulip HTTP client
+     * @param alertWords Add words (or phrases) to the user's set of configured alert words.
+     */
+    public AddAlertWordsApiRequest(ZulipHttpClient client, String[] alertWords) {
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/AddAlertWordsApiRequest.java:26]:
```suggestion
    public AddAlertWordsApiRequest(ZulipHttpClient client, String... alertWords) {
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/AddAlertWordsApiRequest.java [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    /**
+     * Executes the Zulip API request for set alert words.
+     *
+     * @throws ZulipClientException if the request was not successful
+     */
+    @Override
+    public String [] execute() throws ZulipClientException {
+        return client().post(USERS_WITH_ME, getParams(), GetAllAlertWordsApiResponse.class).getAlertWords();
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/AddAlertWordsApiRequest.java:38]: I don't think `USERS_WITH_ME` will work for these API requests. You'll probably need to create a new constant in `UserRequestConstants` like:

```
public static final String USERS_ALERT_WORDS = USERS_WITH_ME + "/alert_words";
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/RemoveAlertWordsApiRequest.java [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+
+    public static final String ALERT_WORDS = "alert_words";
+
+    /**
+     * Constructs a {@link RemoveAlertWordsApiRequest}.
+     *
+     * @param client     The Zulip HTTP client
+     * @param alertWords An array of strings to be removed from the user's set of configured alert words
+     */
+    public RemoveAlertWordsApiRequest(ZulipHttpClient client, String[] alertWords) {
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/RemoveAlertWordsApiRequest.java:26]:
```suggestion
    public RemoveAlertWordsApiRequest(ZulipHttpClient client, String... alertWords) {
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/RemoveAlertWordsApiRequest.java [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+        super(client);
+        putParam(ALERT_WORDS, alertWords);
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/RemoveAlertWordsApiRequest.java:28]:
```suggestion
        putParamAsJsonString(ALERT_WORDS, alertWords);
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/AddAlertWordsApiRequest.java [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * Constructs a {@link AddAlertWordsApiRequest}.
+     *
+     * @param client     The Zulip HTTP client
+     * @param alertWords Add words (or phrases) to the user's set of configured alert words.
+     */
+    public AddAlertWordsApiRequest(ZulipHttpClient client, String[] alertWords) {
+        super(client);
+        putParam(ALERT_WORDS, alertWords);
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/request/AddAlertWordsApiRequest.java:28]:
```suggestion
        putParamAsJsonString(ALERT_WORDS, alertWords);
```

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/response/GetAllAlertWordsApiResponse.java [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.jamesnetherton.zulip.client.api.core.ZulipApiResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Zulip API response class for getting alert words.
+ *
+ * @see <a href="https://zulip.com/api/v1/users/me/alert_words#response">https://zulip.com/api/v1/users/me/alert_words#response</a>
```
Comment (jamesnetherton) [inline src/main/java/com/github/jamesnetherton/zulip/client/api/user/response/GetAllAlertWordsApiResponse.java:12]:
```suggestion
 * @see <a href="https://zulip.com/api/add-alert-words#response">https://zulip.com/api/add-alert-words#response</a>
```

File: src/test/java/com/github/jamesnetherton/zulip/client/api/integration/user/ZulipUserIT.java [hunk 20]
```diff
@@ -470,4 +470,18 @@ public void subGroupManagement() throws Exception {
         zulip.users().deleteUserGroup(groupA.getId()).execute();
         zulip.users().deleteUserGroup(groupB.getId()).execute();
     }
+
+    @Test
+    public void alertWordsManagement() throws Exception {
+        // Create groups
```
Comment (jamesnetherton) [inline src/test/java/com/github/jamesnetherton/zulip/client/api/integration/user/ZulipUserIT.java:476]:
```suggestion
```

**Review by jamesnetherton — COMMENTED** [review]
Thanks a lot for this :+1: 

Sorry for leaving a lot of review comments. It's mostly cosmetic stuff to fix up.

File: src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java (hunk 1 above)
Comment (youknowwjk) [reply src/main/java/com/github/jamesnetherton/zulip/client/api/user/UserService.java]: Very good suggestion, I have already done as you said

**Comment by jamesnetherton** [comment]
Thanks a lot @youknowwjk!

I will try to release 0.5.2 later in the week.

**Pull request merged**